app/os_js.go

```diff
@@ -562,7 +566,12 @@ func (w *window) Configure(options []Option) {
 	w.w.Event(ConfigEvent{Config: w.config})
 }
 
-func (w *window) Perform(system.Action) {}
+func (w *window) Perform(acts system.Action) {
+	if acts&system.ActionClose != 0 {
+		w.w.Event(ViewEvent{})
+		w.w.Event(system.DestroyEvent{})
+	}
+}
```

Comment:
I had expected `Perform(system.ActionClose)` to trigger a shutdown sequence that ended up in, say, the `visibilitychange` listener that takes care of sending `StageEvent` updates. As far as I can tell, everything keeps running even though you've received a `DestroyEvent`.

It would probably be fine to revert this change completely, punting the shutdown implementation.